src/v/http/client.cc

```diff
@@ -589,4 +592,25 @@ ss::input_stream<char> client::response_stream::as_input_stream() {
     return ss::input_stream<char>(std::move(ds));
 }
 
+client::request_header redacted_header(client::request_header original) {
+    using field_type = std::variant<boost::beast::http::field, std::string>;
+
+    static const std::unordered_set<field_type> redacted_fields{
+      boost::beast::http::field::authorization, "x-amz-content-sha256"};
```

Comment:
this might be a useful case to consider using two if statements or a vector with two strings.

the issue with `static` here is that the compiler is going to introduce a bunch of code to protect against multithreaded initialization scenarios.

Reply:
Thanks, I will raise a PR to add this along with the other change and some tests